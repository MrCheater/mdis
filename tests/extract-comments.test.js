const path = require('path');

const { extractComments } = require('../src/extract-comments');

describe('extractComments', () => {
  test('"named-blocks.js"', () => {
    expect(
      extractComments(path.join(__dirname, 'files', 'named-blocks.js'))
    ).toMatchSnapshot();
  });

  test('"simple.js"', () => {
    expect(
      extractComments(path.join(__dirname, 'files', 'simple.js'))
    ).toMatchSnapshot();
  });

  test('"strings-and-comments.js"', () => {
    expect(
      extractComments(path.join(__dirname, 'files', 'strings-and-comments.js'))
    ).toMatchSnapshot();
  });
});
