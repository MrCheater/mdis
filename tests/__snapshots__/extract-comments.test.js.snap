// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractComments "named-blocks.js" 1`] = `
Array [
  Object {
    "source": "mdis-start",
    "string": "mdis-start",
  },
  Object {
    "end": Object {
      "column": 22,
      "line": 1,
    },
    "source": "// mdis-start imports",
    "start": Object {
      "column": 1,
      "line": 1,
    },
    "string": "mdis-start imports",
  },
  Object {
    "end": Object {
      "column": 21,
      "line": 3,
    },
    "source": "// mdis-stop imports",
    "start": Object {
      "column": 1,
      "line": 3,
    },
    "string": "mdis-stop imports",
  },
  Object {
    "end": Object {
      "column": 26,
      "line": 5,
    },
    "source": "// mdis-start NamedBlocks",
    "start": Object {
      "column": 1,
      "line": 5,
    },
    "string": "mdis-start NamedBlocks",
  },
  Object {
    "end": Object {
      "column": 21,
      "line": 6,
    },
    "source": "// mdis-start Sample",
    "start": Object {
      "column": 1,
      "line": 6,
    },
    "string": "mdis-start Sample",
  },
  Object {
    "end": Object {
      "column": 22,
      "line": 8,
    },
    "source": "// mdis-stop Sample",
    "start": Object {
      "column": 3,
      "line": 8,
    },
    "string": "mdis-stop Sample",
  },
  Object {
    "end": Object {
      "column": 26,
      "line": 10,
    },
    "source": "// mdis-start for-i-j",
    "start": Object {
      "column": 5,
      "line": 10,
    },
    "string": "mdis-start for-i-j",
  },
  Object {
    "end": Object {
      "column": 21,
      "line": 13,
    },
    "source": "// some code",
    "start": Object {
      "column": 9,
      "line": 13,
    },
    "string": "some code",
  },
  Object {
    "end": Object {
      "column": 29,
      "line": 14,
    },
    "source": "// mdis-stop for-i-j",
    "start": Object {
      "column": 9,
      "line": 14,
    },
    "string": "mdis-stop for-i-j",
  },
  Object {
    "end": Object {
      "column": 30,
      "line": 16,
    },
    "source": "// mdis-start for-i-j",
    "start": Object {
      "column": 9,
      "line": 16,
    },
    "string": "mdis-start for-i-j",
  },
  Object {
    "end": Object {
      "column": 25,
      "line": 19,
    },
    "source": "// mdis-stop for-i-j",
    "start": Object {
      "column": 5,
      "line": 19,
    },
    "string": "mdis-stop for-i-j",
  },
  Object {
    "end": Object {
      "column": 23,
      "line": 23,
    },
    "source": "// mdis-start render",
    "start": Object {
      "column": 3,
      "line": 23,
    },
    "string": "mdis-start render",
  },
  Object {
    "end": Object {
      "column": 22,
      "line": 27,
    },
    "source": "// mdis-stop render",
    "start": Object {
      "column": 3,
      "line": 27,
    },
    "string": "mdis-stop render",
  },
  Object {
    "end": Object {
      "column": 23,
      "line": 28,
    },
    "source": "// mdis-start Sample",
    "start": Object {
      "column": 3,
      "line": 28,
    },
    "string": "mdis-start Sample",
  },
  Object {
    "end": Object {
      "column": 20,
      "line": 30,
    },
    "source": "// mdis-stop Sample",
    "start": Object {
      "column": 1,
      "line": 30,
    },
    "string": "mdis-stop Sample",
  },
  Object {
    "end": Object {
      "column": 25,
      "line": 31,
    },
    "source": "// mdis-stop NamedBlocks",
    "start": Object {
      "column": 1,
      "line": 31,
    },
    "string": "mdis-stop NamedBlocks",
  },
  Object {
    "source": "mdis-stop",
    "string": "mdis-start",
  },
]
`;

exports[`extractComments "simple.js" 1`] = `
Array [
  Object {
    "source": "mdis-start",
    "string": "mdis-start",
  },
  Object {
    "end": Object {
      "column": 14,
      "line": 1,
    },
    "source": "// mdis-start",
    "start": Object {
      "column": 1,
      "line": 1,
    },
    "string": "mdis-start",
  },
  Object {
    "end": Object {
      "column": 13,
      "line": 3,
    },
    "source": "// mdis-stop",
    "start": Object {
      "column": 1,
      "line": 3,
    },
    "string": "mdis-stop",
  },
  Object {
    "end": Object {
      "column": 14,
      "line": 8,
    },
    "source": "// mdis-start",
    "start": Object {
      "column": 1,
      "line": 8,
    },
    "string": "mdis-start",
  },
  Object {
    "end": Object {
      "column": 13,
      "line": 10,
    },
    "source": "// mdis-stop",
    "start": Object {
      "column": 1,
      "line": 10,
    },
    "string": "mdis-stop",
  },
  Object {
    "source": "mdis-stop",
    "string": "mdis-start",
  },
]
`;

exports[`extractComments "strings-and-comments.js" 1`] = `
Array [
  Object {
    "source": "mdis-start",
    "string": "mdis-start",
  },
  Object {
    "end": Object {
      "column": 3,
      "line": 4,
    },
    "source": "/*
	Multi Line Comment
	Test test test
*/",
    "start": Object {
      "column": 1,
      "line": 1,
    },
    "string": "Multi Line Comment
	Test test test",
  },
  Object {
    "end": Object {
      "column": 23,
      "line": 6,
    },
    "source": "// Single quote string",
    "start": Object {
      "column": 1,
      "line": 6,
    },
    "string": "Single quote string",
  },
  Object {
    "end": Object {
      "column": 23,
      "line": 23,
    },
    "source": "// Double quote string",
    "start": Object {
      "column": 1,
      "line": 23,
    },
    "string": "Double quote string",
  },
  Object {
    "end": Object {
      "column": 20,
      "line": 38,
    },
    "source": "// Back tick string",
    "start": Object {
      "column": 1,
      "line": 38,
    },
    "string": "Back tick string",
  },
  Object {
    "source": "mdis-stop",
    "string": "mdis-start",
  },
]
`;
